go-controller/pkg/ovn/egressip.go

```diff
@@ -861,18 +861,25 @@ func (oc *Controller) syncStaleEgressReroutePolicy(egressIPToPodIPCache map[stri
 	logicalRouterPolicyRes := []nbdb.LogicalRouterPolicy{}
 	opModels := []libovsdbops.OperationModel{
 		{
-			ModelPredicate: func(lrp *nbdb.LogicalRouterPolicy) bool { return lrp.Priority == types.EgressIPReroutePriority },
+			ModelPredicate: func(lrp *nbdb.LogicalRouterPolicy) bool {
+				if lrp.Priority != types.EgressIPReroutePriority {
+					return false
+				}
+				egressIPName := lrp.ExternalIDs["name"]
+				podIPCache, exists := egressIPToPodIPCache[egressIPName]
+				splitMatch := strings.Split(lrp.Match, " ")
+				logicalIP := splitMatch[len(splitMatch)-1]
+				parsedLogicalIP := net.ParseIP(logicalIP)
+				if !exists || !podIPCache.Has(parsedLogicalIP.String()) {
+					klog.Infof("syncStaleEgressReroutePolicy will delete %s: %v", egressIPName, lrp)
+					return true
+				}
+				return false
+			},
 			ExistingResult: &logicalRouterPolicyRes,
 			DoAfter: func() {
 				for _, item := range logicalRouterPolicyRes {
-					egressIPName := item.ExternalIDs["name"]
-					podIPCache, exists := egressIPToPodIPCache[egressIPName]
-					splitMatch := strings.Split(item.Match, " ")
-					logicalIP := splitMatch[len(splitMatch)-1]
-					parsedLogicalIP := net.ParseIP(logicalIP)
-					if !exists || !podIPCache.Has(parsedLogicalIP.String()) {
```

Comment:
this logic needs to move to the predicate! otherwise we would be deleting entries that should not be deleted,
since the LogicalRouterPolicy rows returned are part of the operation, regardless.